theano/tensor/nnet/abstract_conv.py

```diff
@@ -1663,18 +1716,27 @@ def conv(self, img, kern, mode="valid", dilation=1, num_groups=1):
             raise ValueError(
                 'invalid dilation {}, expected {} values'.format(dilation,
                                                                  self.convdim))
+        if unshared and direction == "backprop weights":
+            if mode != "valid":
+                raise ValueError('conv mode for unshared backprop wrt weights must be "valid"')
+            # Do a transpose later to bring it to required shape
+            out_shape = (img.shape[0], kern.shape[0],
+                         kern.shape[2], kern.shape[3],
+                         img.shape[2] - kern.shape[2] + 1,
+                         img.shape[3] - kern.shape[3] + 1)
```

Comment:
Ok to keep it like this if you think it is better. For get_conv_output_shape(), if the output shape is the same for unshared or not, can you add a note in its docstring about that? Otherwise people will think it is just not implemented instead of not needed.

@gvtulder I let you confirm that this is true. I didn't follow all what is related to the shapes computation.